src/Plugin/KeyType/CertificateKeyType.php

```diff
@@ -0,0 +1,170 @@
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\key\Plugin\KeyPluginFormInterface;
+use Drupal\key\Plugin\KeyTypeBase;
+use Drupal\os2web_key\CertificateHelper;
+use Drupal\os2web_key\Exception\RuntimeException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a key that combines a certificate and an optional password.
+ *
+ * @KeyType(
+ *   id = "os2web_key_certificate",
+ *   label = @Translation("Certificate"),
+ *   description = @Translation("A key type to store a certificate."),
+ *   group = "authentication",
+ *   key_value = {
+ *     "plugin" = "textarea_field"
+ *   },
+ * )
+ */
+class CertificateKeyType extends KeyTypeBase implements KeyPluginFormInterface {
+  use DependencySerializationTrait;
+
+  private const PASSPHRASE = 'passphrase';
+  private const INPUT_FORMAT = 'input_format';
+  private const OUTPUT_FORMAT = 'output_format';
+
+  private const FORMAT_PEM = 'pem';
+  private const FORMAT_PFX = 'pfx';
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    array $configuration,
+    $plugin_id,
+    $plugin_definition,
+    private readonly CertificateHelper $certificateHelper,
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition): self {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get(CertificateHelper::class)
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return [
+      self::PASSPHRASE => '',
+      self::INPUT_FORMAT => self::FORMAT_PFX,
+      self::OUTPUT_FORMAT => self::FORMAT_PEM,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form[self::PASSPHRASE] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Passphrase'),
+      '#default_value' => $this->configuration[self::PASSPHRASE] ?? NULL,
+    ];
+
+    $formatOptions = [
+      self::FORMAT_PEM => $this->t('PEM'),
+      self::FORMAT_PFX => $this->t('pfx (p12)'),
+    ];
+    $form[self::INPUT_FORMAT] = [
+      '#title' => $this->t('Input format'),
+      '#type' => 'select',
+      '#options' => $formatOptions,
+      '#required' => TRUE,
+      '#default_value' => $this->configuration[self::INPUT_FORMAT] ?? NULL,
+    ];
+
+    $form[self::OUTPUT_FORMAT] = [
+      '#title' => $this->t('Output format'),
+      '#type' => 'select',
+      '#options' => $formatOptions,
+      '#required' => TRUE,
+      '#default_value' => $this->configuration[self::OUTPUT_FORMAT] ?? NULL,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state): void {
+  }
```

Comment:
Empty validation function, would it be possible to validate the `Passphrase` for the certificate in some way if one is given?

Reply:
Actual validation is performed in `validateKeyValue`. I've added a comment telling this in https://github.com/OS2web/os2web_key/pull/1/commits/0eb16ad83c61f66d80da0e795314dbaf3672216b.